src/lib/utils.py

```diff
@@ -68,6 +68,17 @@ def _extract_timestamp(info):
     raise CheckerRemoteError(f"Cannot parse date/time: {date_str}")
 
 
+def _check_newline(fp):
+    fp.seek(0, os.SEEK_END)
+    fp.seek(fp.tell() - 1, os.SEEK_SET)
+    last_char = fp.read()
+    fp.seek(0, os.SEEK_SET)
```

Comment:
Nitpick: It would nicer to remember the actual initial position and reset to it after we're done, instead of resetting to the beginning.

Reply:
Yeah that'd be safer.